### PR TITLE
Preparation for #1277: refactor rotation scan matcher

### PR DESCRIPTION
**Pull request by schwoere**
Added new constructor for `RotationalScanMatcher` and exposed `RotateHistogram`.

This PR prepares for PR #1277 where the constructor 
`RotationalScanMatcher(const std::vector<std::pair<Eigen::VectorXf, float>>& histograms_at_angles)`
will be removed and only the new constructor
`RotationalScanMatcher(const Eigen::VectorXf& histogram)`
will remain. The unit tests will be updated to use the new constructor in #1277.

### CODE REVIEW

File: cartographer/mapping/internal/3d/scan_matching/rotational_scan_matcher_test.cc [hunk 1]
```diff
@@ -28,7 +28,8 @@ namespace {
 TEST(RotationalScanMatcher3DTest, OnlySameHistogramIsScoreOne) {
   Eigen::VectorXf histogram(7);
   histogram << 1.f, 43.f, 0.5f, 0.3123f, 23.f, 42.f, 0.f;
-  RotationalScanMatcher matcher({{histogram, 0.f}});
+  RotationalScanMatcher matcher(
```
Comment (spielawa) [inline cartographer/mapping/internal/3d/scan_matching/rotational_scan_matcher_test.cc]: Why this change?

Comment (spielawa) [reply cartographer/mapping/internal/3d/scan_matching/rotational_scan_matcher_test.cc]: oh, ic, that's because you'll eventually remove this constructor. I'd move this change to the other PR then, where you actually remove the constructor.

Comment (schwoere) [reply cartographer/mapping/internal/3d/scan_matching/rotational_scan_matcher_test.cc]: Okay, I reverted the change. Now it has the same semantics as before, but I still have to update the tests, otherwise the compiler complains about ambiguous constructors for `RotationalScanMatcher`.

**Comment by schwoere** [comment]
ptal @spielawa

**Comment by schwoere** [comment]
Thanks @spielawa . Can you now have a look as well, @wohe ?

**Comment by wohe** [comment]
@schwoere The test no longer is changed to use the new constructor. Change the PR description to reflect this?

**Comment by schwoere** [comment]
@wohe thanks for the hint, done!

**Comment by schwoere** [comment]
@wally-the-cartographer merge

**Comment by wohe** [comment]
@wally-the-cartographer merge
